app/Features/QueryAnimeSchedulesHandler.php

```diff
@@ -24,7 +25,7 @@ public function __construct(private readonly AnimeRepository $repository)
     public function handle($request)
     {
         $requestParams = collect($request->all());
-        $limit = $requestParams->get("limit");
+        $limit = $request->limit instanceof Optional ? max_results_per_page() : $request->limit;
```

Comment:
This is repeating code, shouldn't we either change the `PreparesData` to replace Optional with this?